wagtailcache/cache.py

```diff
@@ -412,10 +412,9 @@ def _wrapped_view_func(
     ) -> HttpResponse:
         # Try to fetch an already cached page from wagtail-cache.
         response = FetchFromCacheMiddleware().process_request(request)
-        if response:
-            return response
-        # Since we don't have a response at this point, process the request.
-        response = view_func(request, *args, **kwargs)
+        if response is None:
```

Comment:
Looks like this change will reduce performance of the cache reads. Because now, it is fetching from cache (assume the cache is filesystem or network... therefore slow), then instead of immediately returning, it will save that response back into the cache (line 419... once again assume this is very slow). So this will effectively double the time to process a cache read.

Reply:
I don't think that is the case, because the `FetchFromCacheMiddleware` sets `_wagtailcache_update` to `False` on the request, and the `UpdateCacheMiddleware` checks for that and in that case just adjusts headers on the response and returns it without storing it into the cache again on line 290. The same also already happens when using the middleware instead of `cache_page()`

Reply:
Ok, after reading through this makes sense now. Had to wrap my head around it. During normal middleware usage: `FetchFromCacheMiddleware` is called. If it has a "hit", it returns. In which case Django continues calling all the other middlewares, including `UpdateCacheMiddleware`.

So this change makes the `@cache_page` decorator behave more like having both middlewares installed.